components/brave_new_tab/resources/components/settings/background_panel.tsx

```diff
@@ -0,0 +1,198 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Icon from '@brave/leo/react/icon'
+import ProgressRing from '@brave/leo/react/progressRing'
+import Toggle from '@brave/leo/react/toggle'
+
+import { BackgroundType } from '../../models/new_tab_model'
+import { useNewTabModel, useNewTabState } from '../new_tab_context'
+import { useLocale } from '../locale_context'
+import { useCallbackWrapper } from '../../lib/use_callback_wrapper'
+import { inlineCSSVars } from '../../lib/inline_css_vars'
+import { BackgroundTypePanel } from './background_type_panel'
+
+import {
+  backgroundCSSValue,
+  gradientPreviewBackground,
+  solidPreviewBackground } from '../../models/backgrounds'
+
+import { style } from './background_panel.style'
+
+export function BackgroundPanel() {
+  const { getString } = useLocale()
+  const model = useNewTabModel()
+  const wrapCallback = useCallbackWrapper()
+
+  const [
+    backgroundsEnabled,
+    backgroundsCustomizable,
+    sponsoredImagesEnabled,
+    selectedBackgroundType,
+    selectedBackground,
+    braveBackgrounds,
+    customBackgrounds
+  ] = useNewTabState((state) => [
+    state.backgroundsEnabled,
+    state.backgroundsCustomizable,
+    state.sponsoredImagesEnabled,
+    state.selectedBackgroundType,
+    state.selectedBackground,
+    state.braveBackgrounds,
+    state.customBackgrounds
+  ])
+
+  const [panelType, setPanelType] = React.useState<BackgroundType>('none')
+  const [uploading, setUploading] = React.useState(false)
+
+  function getTypePreviewValue(type: BackgroundType) {
+    const isSelectedType = type === selectedBackgroundType
+    switch (type) {
+      case 'brave':
+        return braveBackgrounds.at(0)?.imageUrl || ''
```

Comment:
```suggestion
        return braveBackgrounds.at(0)?.imageUrl ?? ''
```

Reply:
Any reason for preferring `at` here of `[0]`? We're not doing anything fancy like negative indexing

Reply:
Fixed.